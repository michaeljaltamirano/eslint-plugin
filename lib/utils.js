"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isObjectLike = exports.isNotNullish = exports.isJson = void 0;
function isNotNullish(arg) {
    return arg !== null && arg !== undefined;
}
exports.isNotNullish = isNotNullish;
function isObjectLike(arg) {
    return (typeof arg === 'object' && isNotNullish(arg) && !!Object.keys(arg).length);
}
exports.isObjectLike = isObjectLike;
function isJson(arg) {
    if (!isNotNullish(arg))
        return false;
    return isObjectLike(arg);
}
exports.isJson = isJson;
