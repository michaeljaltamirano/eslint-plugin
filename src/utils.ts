function isNotNullish<T>(arg: null | undefined | T): arg is T {
	return arg !== null && arg !== undefined;
}

function isObjectLike(arg: unknown): arg is Record<string, unknown> {
	return (
		typeof arg === 'object' && isNotNullish(arg) && !!Object.keys(arg).length
	);
}

function isJson(
	arg: unknown | Record<string, unknown>,
): arg is Record<string, unknown> {
	if (!isNotNullish(arg)) return false;

	return isObjectLike(arg);
}

export { isJson, isNotNullish, isObjectLike };
