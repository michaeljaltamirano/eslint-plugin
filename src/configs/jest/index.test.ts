import { jest } from '.';
import { generateTest } from '../../tests/utils';

describe('@michaeljaltamirano/eslint-plugin', () => {
	describe('jest config', () => {
		generateTest(__dirname, jest);
	});
});
