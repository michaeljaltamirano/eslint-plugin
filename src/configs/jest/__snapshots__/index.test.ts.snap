// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@michaeljaltamirano/eslint-plugin jest config serializes to a snapshot 1`] = `
Object {
  "env": Object {
    "jest/globals": true,
  },
  "globals": Object {},
  "ignorePatterns": Array [],
  "parserOptions": Object {},
  "plugins": Array [
    "jest",
  ],
  "rules": Object {
    "jest/consistent-test-it": Array [
      "error",
    ],
    "jest/expect-expect": Array [
      "error",
    ],
    "jest/lowercase-name": Array [
      "error",
      Object {
        "ignore": Array [
          "describe",
        ],
        "ignoreTopLevelDescribe": false,
      },
    ],
    "jest/no-alias-methods": Array [
      "error",
    ],
    "jest/no-commented-out-tests": Array [
      "error",
    ],
    "jest/no-conditional-expect": Array [
      "error",
    ],
    "jest/no-deprecated-functions": Array [
      "error",
    ],
    "jest/no-disabled-tests": Array [
      "error",
    ],
    "jest/no-done-callback": Array [
      "error",
    ],
    "jest/no-duplicate-hooks": Array [
      "error",
    ],
    "jest/no-export": Array [
      "error",
    ],
    "jest/no-focused-tests": Array [
      "error",
    ],
    "jest/no-hooks": Array [
      "error",
    ],
    "jest/no-identical-title": Array [
      "error",
    ],
    "jest/no-if": Array [
      "error",
    ],
    "jest/no-interpolation-in-snapshots": Array [
      "error",
    ],
    "jest/no-jasmine-globals": Array [
      "error",
    ],
    "jest/no-jest-import": Array [
      "error",
    ],
    "jest/no-large-snapshots": Array [
      "off",
    ],
    "jest/no-mocks-import": Array [
      "error",
    ],
    "jest/no-restricted-matchers": Array [
      "error",
    ],
    "jest/no-standalone-expect": Array [
      "error",
    ],
    "jest/no-test-prefixes": Array [
      "error",
    ],
    "jest/no-test-return-statement": Array [
      "error",
    ],
    "jest/prefer-called-with": Array [
      "off",
    ],
    "jest/prefer-expect-assertions": Array [
      "off",
    ],
    "jest/prefer-hooks-on-top": Array [
      "error",
    ],
    "jest/prefer-spy-on": Array [
      "error",
    ],
    "jest/prefer-strict-equal": Array [
      "error",
    ],
    "jest/prefer-to-be-null": Array [
      "error",
    ],
    "jest/prefer-to-be-undefined": Array [
      "error",
    ],
    "jest/prefer-to-contain": Array [
      "error",
    ],
    "jest/prefer-to-have-length": Array [
      "error",
    ],
    "jest/prefer-todo": Array [
      "error",
    ],
    "jest/require-to-throw-message": Array [
      "error",
    ],
    "jest/require-top-level-describe": Array [
      "error",
    ],
    "jest/unbound-method": Array [
      "error",
    ],
    "jest/valid-describe": Array [
      "error",
    ],
    "jest/valid-expect": Array [
      "error",
    ],
    "jest/valid-expect-in-promise": Array [
      "error",
    ],
    "jest/valid-title": Array [
      "error",
      Object {
        "ignoreTypeOfDescribeName": true,
      },
    ],
  },
  "settings": Object {},
}
`;
